packages/block-editor/src/components/inserter/index.native.js

```diff
@@ -161,7 +162,30 @@ export class Inserter extends Component {
 	}
 
 	onToggle( isOpen ) {
-		const { onToggle } = this.props;
+		const { blockTypeImpressions, onToggle, updateSettings } = this.props;
+
+		const impressionsRemain = Object.values( blockTypeImpressions ).some(
```

Comment:
What would you think about moving the `impressionsRemain` checking inside of the `!isOpen` check so we only do that work if the inserter is open. The performance gain is probably pretty minimal, so I don't feel strongly.

Reply:
I think that is a solid improvement. Addressed by https://github.com/WordPress/gutenberg/pull/33906/commits/d187cf1696e89253ddcbc8e63b7e53d5ed8aa0af.